HHS/Head-Start-TTADP
# [TTAHUB-3198] Training report alerts

## Description of change
Alerts table shows up on Training Reports landing page below filter button

### Alert types (triggered by unique dates):
- Missing event info (IST Creator or Collaborator) - 20 days past event start date
- Missing session info (IST Creator or Collaborator or POC) - 20 days past session start date
- No sessions created (IST Creator) - 20 days past event start date
- Event not completed (IST Creator or Collaborator) - 20 days past event end date 

### Alert display logic

POC only sees alerts for missing info on session forms where they are listed as POC on the event

IST collaborator sees alerts for missing info on forms where they are listed as Collaborator on the event, or when session needs to be created on events where they are listed as Collaborator

IST creator sees alerts for missing info on forms where they are listed as Collaborator on the event, or when session needs to be created on events where they are listed as Collaborator, AND for events that need to be closed where they are creator.

Alerts remain until the user takes corrective action.

Alert links will go to:
- form missing the information for missing info links
- new session form for "no sessions created" link
- view event page for "event not created" link and there will be a button to complete the event at the top (only for event creator)

### Emails
Added new emails based on the alerts (criterion is the same)
Updated language in existing TR emails


## How to test
Confirm the acceptance criteria are met
Confirm that [the emails](https://docs.google.com/document/d/1FsUqBxr5lqJ4iCjlRhpzZ9Z7Gtl9-0dBVkGWAszBPI8/edit), updated and added, are delivered and the verbiage is correct.

Note that you will need to adjust the schedule used in the cron.js to test the daily emails. You will also obviously need to force TRs/sessions in particular states to get the correct deliverables.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3198
* https://jira.acf.gov/browse/TTAHUB-3199

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
